ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +82,71 @@ tests = testGroup "Mempool"
   , testProperty "removeTxs"                               prop_Mempool_removeTxs
   ]
 
+{-------------------------------------------------------------------------------
+  Mempool Pure Properties
+-------------------------------------------------------------------------------}
+-- | Test that removed transactions are not in the
+--   result transactionsIds of the returned internal state.
+prop_Mempool_pureRemoveTxsId_removed
+  :: TestSetupWithTxsAndSubTxs
+  -> Property
+prop_Mempool_pureRemoveTxsId_removed TestSetupWithTxsAndSubTxs {..} =
+    withInternalState (testSetup setupWithTxs) $
+    \ mpArgs internalSt ledgerState ->
+      let testTxs               = map fst (txs setupWithTxs)
+          testTxsTickets        = map (\tx -> TxTicket tx (TicketNo 0) 0) testTxs
+          testTxsIds            = map txId testTxs
+          testTxsAll            = testTxsTickets ++ toList (isTxs internalSt)
+          testTxsIdsAll         = testTxsIds ++ Set.toList (isTxIds internalSt)
+          internalSt'           = internalSt {isTxs = fromList testTxsAll,
+                                                 isTxIds = Set.fromList testTxsIdsAll}
+          txIdsToRemove         = map (txId . fst) subTxs
+          -- The function to test: pureRemoveTxs
+          (internalStRes, _)    = pureRemoveTxs txIdsToRemove mpArgs internalSt' ledgerState
+          txIdsRemaining        = Set.toList (isTxIds internalStRes)
+      in property $ null $ intersect txIdsRemaining txIdsToRemove
+
+data TestSetupWithTxsAndSubTxs = TestSetupWithTxsAndSubTxs
+  { setupWithTxs :: TestSetupWithTxs
+  , subTxs       :: [(TestTx, Bool)]
+    -- ^ The 'Bool' indicates whether the transaction is valid
+  } deriving (Show)
```

Comment:
Datatype with:

* An initial ledger state
* A list of initial transactions that are going to be the initial contents of the Mempool
* A list containing a subset of those initial transactions that can be used for `pureRemoveTxs`.

Write a function that takes that data type and produces the corresponding `InternalState`